clang/docs/LanguageExtensions.rst

```diff
@@ -1958,6 +1958,24 @@ references can be used instead of numeric references.
       return -1;
   }
 
+
+Constexpr strings in GNU ASM statememts
+=======================================
+
+In C++11 mode (and greater), Clang supports specifying the template,
+constraints, and clobber strings with a parenthesized constant expression
+producing an object with ``data`` and ``size`` member functions,
+such as ``std::string``.
```

Comment:
to be coy, it might be nice to list `std::string_view` and `std::vector` here too (or just `std::vector<char>`?).  There are ALSO constraints based on whether `data` returns a char type or `size` returns an integral type, right? We should probably mention that.

Reply:
I'd rather reference the standard, but say something along the lines of "follow the same rules as static_assert message in C++2<whatever>, see `[dcl.pre] p12`.